pkg/dbmate/migration.go

```diff
@@ -102,9 +104,9 @@ func parseMigrationContents(contents string) (*ParsedMigration, error) {
 	downBlock := substring(contents, downDirectiveStart, len(contents))
 
 	parsed := ParsedMigration{
-		Up:          upBlock,
+		Up:          splitSQLStatements(upBlock),
 		UpOptions:   parseMigrationOptions(upBlock),
-		Down:        downBlock,
+		Down:        splitSQLStatements(downBlock),
 		DownOptions: parseMigrationOptions(downBlock),
```

Comment:
This approach isn't properly handling options on each migration.

Implementing this feature correctly will require a redesign of the `parseMigrationContents()` from the current naive approach to one that splits the file into regions and then parses each region, filling arrays as it goes.

Reply:
should be fine by now

Reply:
Could you add a test that would otherwise fail if the proper support is not implemented correctly?

Reply:
the implementation is the same as in main branch, no changes here